### PR TITLE
SumBoundary: Use Fewer Guard Cells for J/rho

### PR DESCRIPTION
**Pull request by EZoni**
Exchange less guard cells for the J and rho MultiFabs, depending on the number of guard cells used for current and charge deposition as well as a number of additional operations (e.g. filtering, centering, etc.).

### CODE REVIEW

File: Source/Parallelization/WarpXSumGuardCells.H [hunk 1]
```diff
@@ -66,7 +68,7 @@ WarpXSumGuardCells(amrex::MultiFab& dst, amrex::MultiFab& src,
     else  // Update only the valid cells
         n_updated_guards = amrex::IntVect::TheZeroVector();
 
-    src.SumBoundary(0, ncomp, n_updated_guards, period);
+    src.SumBoundary(0, ncomp, src_ngrow, n_updated_guards, period);
     amrex::Copy( dst, src, 0, icomp, ncomp, n_updated_guards );
```
Comment (WeiqunZhang) [inline Source/Parallelization/WarpXSumGuardCells.H]:
```suggestion
    dst.setVal(0., icomp, ncomp, n_updated_guards);
    dst.ParallelAdd(src, 0, icomp, ncomp, src_ngrow, n_updated_guards, period); 
```

**Comment by EZoni** [comment]
@RemiLehe Actually the change in https://github.com/ECP-WarpX/WarpX/pull/2323/commits/4060bb13a6717b0602c9cd7eecd45c3420364941, where we take into account the stencil of the bilinear filter in order to compute the `src_ngrow` to be passed eventually to `SumBoundary`, caused many more CI tests to fail, compared to the initial commit https://github.com/ECP-WarpX/WarpX/pull/2323/commits/a204ea4fe4487a5c4f1c3a63110c9ee041e58caa, where we had not taken that stencil into account. 

Update: I will fix the implementation, the issue is related to the fact that `ng_depos_J` was increased without being reset to the correct initial value within the loop over `idim`. Should be now fixed in https://github.com/ECP-WarpX/WarpX/pull/2323/commits/758a699a2d51caebc29e3f454fba02ab5fc21640.

File: Source/Parallelization/WarpXSumGuardCells.H (hunk 1 above)
Comment (EZoni) [reply Source/Parallelization/WarpXSumGuardCells.H]: Thank you, Weiqun! Done in https://github.com/ECP-WarpX/WarpX/pull/2323/commits/76935032558f5f6115c56e5ebdc4e8f01499008f, together with the reversion of https://github.com/ECP-WarpX/WarpX/commit/4060bb13a6717b0602c9cd7eecd45c3420364941, due to the reasons explained in https://github.com/ECP-WarpX/WarpX/pull/2323#issuecomment-924448645.

**Comment by EZoni** [comment]
Notes on CI status:
- The benchmark comparisons for `averaged_galilean_2d_psatd_hybrid` fail with a maximum relative error of order O(1e-5) on Ex, By, jz.
- The benchmark comparisons for `averaged_galilean_3d_psatd_hybrid` fail with a maximum relative error of order O(1e-5) on jz.

Update: the stencil of the current centering operation is now taken into account with https://github.com/ECP-WarpX/WarpX/pull/2323/commits/0448bcc8e18bf1c8bddf587511d872fcd6515abc, thus there is no need to reset these benchmarks.

**Review by WeiqunZhang — APPROVED** [review]

**Comment by ax3l** [comment]
Thank you both!

**Review by ax3l — APPROVED** [review]

**Pull request merged**